test/parallel/test-timers-immediate-as-number.js

```diff
@@ -0,0 +1,10 @@
+'use strict';
+const common = require('../common');
+const { test } = require('node:test');
+
+test('setImmediate return value is coerced to a number', (t) => {
+  const immediateId = +setImmediate(common.mustNotCall());
+  clearImmediate(immediateId);
+
+  t.assert.ok(!Number.isNaN(immediateId), 'setImmediate return value is NaN');
```

Comment:
Nit: `!isNaN()` without prior `typeof` check might be not failproof.
```suggestion
  t.assert.strictEqual(typeof immediateId, 'number');
  t.assert.ok(!Number.isNaN(immediateId), 'setImmediate return value is NaN');
```